examples/flutter_gallery/lib/gallery/backdrop.dart

```diff
@@ -0,0 +1,329 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/material.dart';
+
+const double _kFrontHeadingHeight = 32.0; // front layer beveled rectangle
+const double _kFrontClosedHeight = 72.0; // front layer height when closed
+const double _kBackAppBarHeight = 56.0; // back layer (options) appbar height
+
+// The size of the front layer heading's left and right beveled corners.
+final Tween<BorderRadius> _kFrontHeadingBevelRadius = new BorderRadiusTween(
+  begin: const BorderRadius.only(
+    topLeft: const Radius.circular(12.0),
+    topRight: const Radius.circular(12.0),
+  ),
+  end: const BorderRadius.only(
+    topLeft: const Radius.circular(_kFrontHeadingHeight),
+    topRight: const Radius.circular(_kFrontHeadingHeight),
+  ),
+);
+
+class _IgnorePointerWhileStatusIsNot extends StatefulWidget {
+  const _IgnorePointerWhileStatusIsNot({
+    Key key,
+    this.controller,
+    this.status,
+    this.child,
+  }) : super(key: key);
+
+  final AnimationController controller;
+  final AnimationStatus status;
+  final Widget child;
+
+  @override
+  _IgnorePointerWhileStatusIsNotState createState() => new _IgnorePointerWhileStatusIsNotState();
+}
+
+class _IgnorePointerWhileStatusIsNotState extends State<_IgnorePointerWhileStatusIsNot> {
+  bool _ignoring;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.controller.addStatusListener(_handleStatusChange);
+    _ignoring = widget.controller.status != AnimationStatus.completed;
+  }
+
+  @override
+  void dispose() {
+    widget.controller.removeStatusListener(_handleStatusChange);
+    super.dispose();
+  }
+
+  void _handleStatusChange(AnimationStatus _) {
+    final bool value = widget.controller.status != widget.status;
+    if (_ignoring != value) {
+      setState(() {
+        _ignoring = value;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new IgnorePointer(
+      ignoring: _ignoring,
+      child: widget.child,
+    );
+  }
+}
+
+class _CrossFadeTransition extends AnimatedWidget {
+  const _CrossFadeTransition({
+    Key key,
+    this.alignment: Alignment.center,
+    Animation<double> progress,
+    this.child0,
+    this.child1,
+  }) : super(key: key, listenable: progress);
+
+  final AlignmentGeometry alignment;
+  final Widget child0;
+  final Widget child1;
+
+  @override
+  Widget build(BuildContext context) {
+    final Animation<double> progress = listenable;
+
+    final double opacity1 = new CurvedAnimation(
```

Comment:
I suspect you don't have to keep constructing these right? You can just have a const curve or 2 and call `transform` on progress.value

Reply:
They depend on the listenable progress parameter.